src/ol/geom/Geometry.js

```diff
@@ -170,6 +175,14 @@ class Geometry extends BaseObject {
   getExtent(opt_extent) {
     if (this.extentRevision_ != this.getRevision()) {
       this.extent_ = this.computeExtent(this.extent_);
+      if (
+        !containsExtent(
+          [-Infinity, -Infinity, Infinity, Infinity],
+          this.extent_
+        )
+      ) {
+        this.extent_ = createEmpty();
+      }
```

Comment:
```suggestion
      const extent = this.computeExtent(this.extent_);
      if (extent[0] === NaN || extent[1] === NaN) {
        createOrUpdateEmpty(extent);
      }
```
would be more efficient and create less garbage.

Reply:
done
(although `=== NaN` doesn't work as `NaN === NaN` returns false)

Reply:
Right, sorry for that. I meant `isNaN(extent[0])`